docs/DeveloperGuide.md

```diff
@@ -1,38 +1,461 @@
+* `execute`: Method to add the new product to the product list.
+* `createProduct`: Method to create a new product with a unique product ID.
+
+The following sequence diagram shows how an add operation works when the user inputs a valid `add` command:
+<img src="images/AddCommandSequence.png" alt="AddCommandSequence.png"/>
+
+The following sequence diagram details how `AddCommand#execute()` functions.
+<img src="images/AddCommandExecuteSequenceDiagram.png" alt="AddCommandExecuteSequenceDiagram.png"/>
+
+
+### Edit product feature
+#### Implementation
+The `edit` command is used to edit product details such as name, quantity, price and description.
+
+The edit product feature is facilitated by `EditCommand` which extends `Command`.
+
+The following sequence diagram summarizes what happens when a user inputs a valid `edit` command.
+<img src="images/EditCommandSequenceDiagram.png" alt="EditCommandSequenceDiagram.png"/>
+
+> INFO:
+> The lifeline for EditCommand should end at the destroy marker (X) but due to a limitation of PlantUML, 
+> the lifeline reaches the end of diagram.
+
+Validation of the user input is done in `Parser`, hence `EditCommand` assumes that all fields provided upon creation
+of a `EditCommand` object are properly formatted.
+
+Validation for other criteria are still carried out within `EditCommand`.
+1. Checking if at least 1 field (name, quantity, price or description) is provided.
+2. Checking if the product ID (PID) belongs to an existing product.
+
+Once all validation is completed, updating of product details is done by calling `ProductList#updateProduct()`.
+
+The following sequence diagram details how `EditCommand#execute()` functions.
+<img src="images/EditCommandExecuteSequenceDiagram.png" alt="EditCommandExecuteSequenceDiagram.png"/>
+
+**Aspect: Validating parameters and handling errors**
+
+- Alternative 1 (current choice): Check parameters and handle errors within `EditCommand`.
+  - Pros: Easy to implement
+  - Cons: -
+  
+- Alternative 2: Handle validation of errors within `productList#updateProduct()`.
+    - Pros: Implementing `EditCommand#execute()` will be very simple. Usage of throw/catch to handle errors.
+    - Cons: `productList#updateProduct()` will be more lengthy. May require further abstraction.
+
+### List feature
+<img src="images/ListCommandClass.png" alt=""/>
+
+The ListCommand class is responsible for sorting and printing out the products in the list. 
+
+**Attributes**
+* sortType: Additional optional flags the user can set to sort the products in the list.
+
+**Methods**
+* `ListCommand`: Constructor for creating a new instance of the ListCommand class.
+* `execute`: Method to list out the products in the product list.
+* `sortListAccordingly`: Method to sort the list according to the products' PID, products' name or products' quantity.
+
+The following sequence diagram shows how a list operation works, by calling `list`.
+<img src="images/ListCommandSequence.png" alt=""/>
```

Comment:
Sequence diagram is straight to the point good job. Appreciate the use of different colour arrows depending on which block they originate from. Only thing to keep in mind is to ensure that the arrows are still visible as they are quite thin!